models/lpjguess/R/met2model.LPJGUESS.R

```diff
@@ -0,0 +1,139 @@
+# which accompanies this distribution, and is available at
+# http://opensource.ncsa.illinois.edu/license.html
+#-------------------------------------------------------------------------------
+#
+## R Code to convert NetCDF CF met files into LPJ-GUESS met files
+
+##If files already exist in "Outfolder", the default function is NOT to overwrite them and 
+##only gives user the notice that file already exists. If user wants to overwrite the existing files, just change 
+##overwrite statement below to TRUE.
+
+##' met2model wrapper for LPJ-GUESS
+##'
+##' @title met2model.LPJGUESS
+##' @export
+##' @param in.path location on disk where inputs are stored
+##' @param in.prefix prefix of input and output files
+##' @param outfolder location on disk where outputs will be stored
+##' @param start_date the start date of the data to be downloaded (will only use the year part of the date)
+##' @param end_date the end date of the data to be downloaded (will only use the year part of the date)
+##' @param overwrite should existing files be overwritten
+##' @param verbose should the function be very verbose
+##' @author Istem Fer
+met2model.LPJGUESS <- function(in.path, in.prefix, outfolder, start_date, end_date, ..., overwrite=FALSE,verbose=FALSE){
+  
+  library(PEcAn.utils)
+  require(ncdf4)
+  require(lubridate)
+  
+  print("START met2model.LPJGUESS")
+  start_date <- as.POSIXlt(start_date, tz = "GMT")
+  end_date <- as.POSIXlt(end_date, tz = "GMT")
+  start_year <- year(start_date)
+  end_year <- year(end_date)
+  
+  year = sprintf("%04d",seq(start_year,end_year,1))
+  nyear = length(year) #number of years to simulate
+  
+  ## LPJ-GUESS looks for different input files for different climate variables
+  out.files.full <- list()
+  var.names=c("tmp","pre","cld")
+  n.var=length(var.names)
+  long.names=c("air_temperature","precipitation_flux","surface_downwelling_shortwave_flux_in_air")
+  for(i in 1:n.var) out.files.full[[i]] <-file.path(outfolder,paste(in.prefix, start_year, end_year, var.names[[i]], "nc", sep="."))
+  
+  results <- data.frame(file = unlist(out.files.full),
+                        host = fqdn(),
+                        mimetype ='application/x-netcdf',
+                        formatname = 'lpj-guess.metfile' ,
+                        startdate = start_date ,
+                        enddate = end_date,
+                        dbfile.name = unlist(out.files.full),
+                        stringsAsFactors = FALSE)
+  print("internal results")
+  print(results)
+  
+  
+  ## check to see if the outfolder is defined, if not create directory for output
+  if(!file.exists(outfolder)){
+    dir.create(outfolder)
+  }
+
+    ## open netcdf files
+    ncin <- lapply(file.path(in.path,paste(in.prefix,year,"nc",sep=".")),nc_open)
+    
+    ## retrieve lat/lon 
+    lon=ncvar_get(ncin[[1]],"longitude")
+    lat=ncvar_get(ncin[[1]],"latitude")
+    
+    ## at least 2 lat-lon required for LPJ-GUESS to load the data
+    lon=c(lon,lon)
```

Comment:
this and writing the same climate data four times for all (same) four grids below are the hacks that I could have come up with to make it work for LPJ-GUESS